packages/replay-next/src/suspense/SearchCache.ts

```diff
@@ -99,13 +117,32 @@ export const searchCache = createStreamingCache<
       return;
     }
 
-    const sourceIds = await getSourceIds(replayClient, includeNodeModules);
+    const includedFilesMatcher = new Minimatch(includedFiles, { matchBase: true });
+    const excludedFilesMatcher = new Minimatch(excludedFiles, { matchBase: true });
+    const sourceIds = await getSourceIds(replayClient, source => {
+      if (!source.url) {
+        return false;
+      }
+
+      if (openSourceIds && !openSourceIds.includes(source.sourceId)) {
+        return false;
+      }
+
+      if (defaultFilter && (isModuleFromCdn(source) || isNodeModule(source))) {
+        return false;
+      }
+
+      return (
+        (!includedFiles || includedFilesMatcher.match(source.url)) &&
+        !excludedFilesMatcher.match(source.url)
+      );
```

Comment:
nit: I think this could be written a little more clearly if you split it up:

```js
if (excludedFiles && excludedFilesMatcher.match(source.url)) {
  return false;
}

if (includedFiles && !includedFilesMatcher.match(source.url)) {
  return false;
}

return true;
```